ts/src/ellipx.ts

```diff
@@ -0,0 +1,1583 @@
+        const token = this.safeValue (currency, 'Crypto_Token', {});
+        const code = this.safeCurrencyCode (this.safeString (token, 'Symbol'));
+        const name = this.safeString (token, 'Name');
+        const active = this.safeString (currency, 'Status') === 'valid';
+        const deposit = this.safeString (currency, 'Can_Deposit') === 'Y';
+        const withdraw = this.safeString (currency, 'Status') === 'valid';
+        let fee = undefined;
+        if (currency['Withdraw_Fee'] !== undefined) {
+            fee = this.parseAmount (currency['Withdraw_Fee']);
+        }
+        const precision = this.parseNumber (this.parsePrecision (this.safeString (token, 'Decimals')));
+        let minDeposit = undefined;
+        if (currency['Minimum_Deposit'] !== undefined) {
+            minDeposit = this.parseAmount (currency['Minimum_Deposit']);
+        }
+        let minWithdraw = undefined;
+        if (currency['Minimum_Withdraw'] !== undefined) {
+            minWithdraw = this.parseAmount (currency['Minimum_Withdraw']);
+        }
+        const networkId = this.safeString (currency, 'Crypto_Chain__');
+        const networkData = this.safeValue (currency, 'Crypto_Chain', {});
+        const networkCode = this.safeString (networkData, 'Type', 'default');
+        const networks = {
+            'string': undefined,
+            'info': networkCode === 'default' ? {} : networkData,
+            'id': networkId || id || '',
+            'network': networkCode,
+            'active': active,
+            'deposit': deposit,
+            'withdraw': withdraw,
+            'fee': fee,
+            'precision': precision,
+            'limits': {
+                'deposit': {
+                    'min': minDeposit,
+                    'max': undefined,
+                },
+                'withdraw': {
+                    'min': minWithdraw,
+                    'max': undefined,
+                },
+            },
+        };
+        const result: Currency = {
+            'info': currency,
+            'id': id,
+            'code': code,
+            'name': name,
+            'active': active,
+            'deposit': deposit,
+            'withdraw': withdraw,
+            'fee': fee,
+            'precision': precision,
+            'type': undefined,
+            'limits': {
+                'amount': {
+                    'min': undefined,
+                    'max': undefined,
+                },
+                'withdraw': {
+                    'min': minWithdraw,
+                    'max': undefined,
+                },
+            },
+            'networks': networks,
+        };
+        return result;
+    }
+
+    /**
+     * @method
+     * @name ellipx#fetchTrades
+     * @description fetches all completed trades for a particular market/symbol
+     * @param {string} symbol unified market symbol (e.g. 'BTC/USDT')
+     * @param {int|undefined} since timestamp in ms of the earliest trade to fetch
```

Comment:
that's not the standard way of saying the param is optional, instead we should do `@param {int} [since]` (using the brackets)